addon/routes/flexberry-map.js

```diff
@@ -1,62 +1,21 @@
-import Ember from 'ember';
+/**
+  @module ember-flexberry-gis
+ */
+
 import EditFormRoute from 'ember-flexberry/routes/edit-form';
-import QueryBuilder from 'ember-flexberry-data/query/builder';
-import FilterOperator from 'ember-flexberry-data/query/filter-operator';
+import FlexberryMapLoadLayersMixin from 'ember-flexberry-gis/mixins/flexberry-map-load-layers';
 
-export default EditFormRoute.extend({
+/**
+  Route for show map page, in addition to model also load layers hierarhy
+  @class FlexberryMapRoute
+  @extends EditFormRoute
+  @uses FlexberryMapLoadLayersMixin
+ */
+export default EditFormRoute.extend(FlexberryMapLoadLayersMixin, {
   modelProjection: 'MapE',
   modelName: 'new-platform-flexberry-g-i-s-map',
 
-  loadLayer(layer) {
-    return new Ember.RSVP.Promise((resolve, reject) => {
-      if (layer.get('type') === 'group') {
-        this.loadChildLayers(layer)
-          .then(layers => layer.set('layers', layers))
-          .then(() => resolve(layer))
-          .catch(reason => reject(reason));
-      }
-      else {
-        resolve(layer);
-      }
-    });
-  },
-
-  loadChildLayers(layer) {
-    let queryModel = 'new-platform-flexberry-g-i-s-map-layer';
-
-    let query =
-      new QueryBuilder(this.store)
-        .from(queryModel)
-        .selectByProjection('MapLayerE')
-        .where('parent', FilterOperator.Eq, layer.id);
-
-    return this.store
-      .query(queryModel, query.build())
-      .then(queryLayers => {
-        let promises = queryLayers.map(lyr => this.loadLayer(lyr));
-        return Ember.RSVP.Promise.all(promises);
-      });
-  },
-
-  setIndex(layers, indexed) {
-    if (layers) {
-      layers.forEach(layer => {
-        layer.set('index', indexed.index);
-        indexed.index++;
-        this.setIndex(layer.get('layers'), indexed);
-      });
-    }
-  },
-
   model() {
-    let baseModel = null;
-    return this._super(...arguments).then(map => {
-      baseModel = map;
-      return this.loadChildLayers(map.get('rootLayer'));
-    }).then(layers => {
-      baseModel.set('layers', layers);
-      this.setIndex(layers, { index: 0 });
-      return baseModel;
-    });
+    return this.loadMapLayers(this._super(...arguments));
```

Comment:
Don't you think that separate calls would be clearer:
let mapPromise = this._super(...arguments);
return this.loadMapLayers(mapPromise);